### PR TITLE
Qmaps 666 poi direction

### PR DESCRIPTION
**Pull request by Julien-laville**
Create link for direction from poi

### CODE REVIEW

File: src/views/poi_panel.dot [hunk 1]
```diff
@@ -8,12 +8,18 @@
...
        {{= this.minimalHourPanel.render() }}
       </div>
+      <div class="poi_panel__content__card__action__container">
         <button class="poi_panel__content__card__action" {{= click(this.showDetail, this) }}>
           <span class="icon-chevrons-right"></span>
           {{= _('SEE MORE', 'poi panel') }}
         </button>
+        <button class="poi_panel__content__card__action poi_panel__content__card__action__direction" {{= click(this.openDirection, this) }}>
+          <span class="icon-chevrons-right"></span>
+          {{= _('DIRECTION', 'poi panel') }}
```
Comment (amatissart) [inline src/views/poi_panel.dot]: * wrong icon
* directions button should be the first one
* `'DIRECTION'` :arrow_right: `'DIRECTIONS'`

File: src/views/poi_panel.dot [hunk 2]
```diff
@@ -32,6 +38,11 @@
           </button>
           <div class="poi_panel__actions__text">{{= _('SHARE', 'poi') }}</div>
         </div>
+        <div class="poi_panel__actions__direction_container" {{= click(this.openDirection, this) }}>
+          <button class="poi_panel__actions__icon icon-corner-up-right">
+          </button>
+          <div class="poi_panel__actions__text">{{= _('DIRECTION', 'poi') }}</div>
```
Comment (amatissart) [inline src/views/poi_panel.dot]: `'DIRECTION'` :arrow_right: `'DIRECTIONS'`

File: src/adapters/scene_direction.js [hunk 3]
```diff
@@ -122,6 +122,11 @@ export default class SceneDirection {
       this.map.getSource(`source_${route.id}`).setData(this.buildRouteData([]))
```
Comment (amatissart) [inline src/adapters/scene_direction.js]: I think these empty coordinates `[]` may cause weird behavior of mapbox-gl-js ?  
Why not simply use `removeLayer` here ?

File: src/views/poi_panel.dot [hunk 4]
```diff
@@ -8,12 +8,20 @@
...
         <button class="poi_panel__content__card__action" {{= click(this.showDetail, this) }}>
           <span class="icon-chevrons-right"></span>
           {{= _('SEE MORE', 'poi panel') }}
         </button>
+        {{? this.isDirectionActive }}
+          <button class="poi_panel__content__card__action poi_panel__content__card__action__direction" {{= click(this.openDirection, this) }}>
+            <span class="icon-chevrons-right"></span>
+            {{= _('DIRECTIONS', 'poi panel') }}
+          </button>
+        {{?}}
```
Comment (amatissart) [inline src/views/poi_panel.dot]: These issues have not been corrected:
* wrong icon
* directions button should be the first one

Also, I am not sure about the alignment of these buttons. 
Shouldn't they be put besides the "close" button ?

File: src/adapters/scene_direction.js [hunk 5]
```diff
@@ -50,15 +50,15 @@ export default class SceneDirection {
 
   showMarkerSteps() {
     if(this.vehicle !== "walking" && window.innerWidth > 640) {
-      for (let step in this.steps) {
+      this.steps.forEach((step) => {
         const markerStep = document.createElement('div')
         markerStep.className = 'itinerary_marker_step'
         this.markersSteps.push(
           new Marker(markerStep)
             .setLngLat(this.steps[step].maneuver.location)
```
[the remark this answers is not in the document]
Comment (Julien-laville) [reply src/adapters/scene_direction.js]: : ]

File: src/adapters/scene_direction.js (hunk 3 above)
Comment (Julien-laville) [reply src/adapters/scene_direction.js]: You are right, i'll remove the source (now i know it's possible) then remove the "update data source" case on set route

**Pull request merged**